lib/blockscore/actions/retrieve.rb

```diff
@@ -12,7 +12,7 @@ module Actions
     module Retrieve
       module ClassMethods
         def retrieve(id, options = {})
-          fail ArgumentError if id.empty?
+          fail ArgumentError,'ID must be supplied' if id.nil? || id.empty?
```

Comment:
This is a tiny preference, but I would typically have a trailing space after the comma.